src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1024,16 +1025,25 @@ public long getIdLong()
 
     @Nonnull
     @Override
-    public AuditableRestAction<Integer> prune(int days, boolean wait)
+    public AuditableRestAction<Integer> prune(int days, boolean wait, @Nonnull Role... roles)
     {
         checkPermission(Permission.KICK_MEMBERS);
 
         Checks.check(days >= 1 && days <= 30, "Provided %d days must be between 1 and 30.", days);
+        Checks.notNull(roles, "Roles");
 
-        Route.CompiledRoute route = Route.Guilds.PRUNE_MEMBERS.compile(getId()).withQueryParams("days", Integer.toString(days));
+        Route.CompiledRoute route = Route.Guilds.PRUNE_MEMBERS.compile(getId());
+        FormBody.Builder form = new FormBody.Builder();
+        form.add("days", Integer.toString(days));
         if (!wait)
-            route = route.withQueryParams("compute_prune_count", "false");
-        return new AuditableRestActionImpl<>(getJDA(), route, (response, request) -> response.getObject().getInt("pruned", 0));
+            form.add("compute_prune_count", "false");
+        for (Role role : roles)
+        {
+            Checks.notNull(role, "Role");
+            Checks.check(role.getGuild().equals(this), "Role is not from the same guild!");
+            form.add("include_roles", role.getId());
```

Comment:
The way this works is basically building a form-body in this format:

```
days=30&compute_prune_count=false&include_roles=123&include_roles=456&include_roles=789
``` 

This is the correct way to pass an array through a query format, in the case of the discord API (which uses Flask).